samples/nrf9160/modem_shell/src/rest/rest_shell.c

```diff
@@ -265,17 +277,35 @@ int rest_shell(const struct shell *shell, size_t argc, char **argv)
 	if (ret) {
 		mosh_error("Error %d from rest_lib", ret);
 	} else {
+		int header_len = resp_ctx.response - req_ctx.resp_buff;
+		char *header_start = req_ctx.resp_buff;
+
+		if (print_header_from_resp) {
+			resp_headers = k_calloc(header_len + 1, sizeof(char));
```

Comment:
I think you could get rid of making a copy of the headers by using `%.*s` and `header_len` when printing the headers. See here for an example: https://github.com/nrfconnect/sdk-nrf/commit/1378fb29ba22818d9e4e4f6e5c3233ca2bfa45d9

Reply:
Thanks. Good trick. Changed accordingly.